rules/framework.bzl

```diff
@@ -241,54 +254,57 @@ def _apple_framework_packaging_impl(ctx):
-                framework_root,
-                "--inputs",
-                ctx.actions.args().use_param_file("%s", use_always = True).set_param_file_format("multiline")
-                    .add_all(framework_files),
-                "--outputs",
-                framework_manifest.path,
-            ],
-            outputs = [framework_manifest],
-            mnemonic = "CleaningFramework",
-            execution_requirements = {
-                "local": "True",
-            },
-        )
-    else:
-        ctx.actions.write(framework_manifest, "# Empty framework\n")
+    compilation_context_fields = {}
+
+    if not virtualize_frameworks:
+        framework_files = _concat(binary_out, modulemap_out, header_out, private_header_out, swiftmodule_out, swiftdoc_out)
+        framework_root = _find_framework_dir(framework_files)
+        if framework_root:
+            ctx.actions.run(
+                executable = ctx.executable._framework_packaging,
+                arguments = [
+                    "--action",
+                    "clean",
+                    "--framework_name",
+                    framework_name,
+                    "--framework_root",
+                    framework_root,
+                    "--inputs",
+                    ctx.actions.args().use_param_file("%s", use_always = True).set_param_file_format("multiline")
+                        .add_all(framework_files),
+                    "--outputs",
+                    framework_manifest.path,
+                ],
+                outputs = [framework_manifest],
+                mnemonic = "CleaningFramework",
+                execution_requirements = {
+                    "local": "True",
+                },
+            )
+            compilation_context_fields["framework_includes"] = depset(
+                direct = [paths.dirname(framework_root)],
+            )
+        else:
+            ctx.actions.write(framework_manifest, "# Empty framework\n")
+
+    # Gather objc provider fields
+    # Eventually we need to remove any reference to objc provider
+    # and use CcInfo instead, see this issue for more details: https://github.com/bazelbuild/bazel/issues/10674
 
-    # gather objc provider fields
     objc_provider_fields = {
         "providers": [dep[apple_common.Objc] for dep in ctx.attr.transitive_deps],
     }
-    compilation_context_fields = {}
 
-    if framework_root:
-        compilation_context_fields["framework_includes"] = depset(
-            direct = [paths.dirname(framework_root)],
-        )
     _add_to_dict_if_present(compilation_context_fields, "headers", depset(
         direct = header_out + private_header_out + modulemap_out,
     ))
+
     _add_to_dict_if_present(compilation_context_fields, "defines", depset(
         direct = [],
         transitive = [getattr(dep[CcInfo].compilation_context, "defines") for dep in ctx.attr.deps if CcInfo in dep],
     ))
-    _add_to_dict_if_present(objc_provider_fields, "module_map", depset(
```

Comment:
We need to move this to line 287, right?

Reply:
Yep - this is moved to another line